test/object-store/util/sync/sync_test_utils.cpp

```diff
@@ -172,18 +172,38 @@ ExpectedRealmPaths::ExpectedRealmPaths(const std::string& base_path, const std::
 
 #if REALM_ENABLE_AUTH_TESTS
 
+static std::string unquote_string(std::string_view possibly_quoted_string)
+{
+    if (possibly_quoted_string.size() > 0) {
+        auto check_char = possibly_quoted_string.front();
+        if (check_char == '"' || check_char == '\'') {
+            possibly_quoted_string.remove_prefix(1);
+        }
+    }
+    if (possibly_quoted_string.size() > 0) {
+        auto check_char = possibly_quoted_string.back();
+        if (check_char == '"' || check_char == '\'') {
+            possibly_quoted_string.remove_suffix(1);
+        }
+    }
+    return std::string{possibly_quoted_string};
+}
+
 #ifdef REALM_MONGODB_ENDPOINT
 std::string get_base_url()
 {
     // allows configuration with or without quotes
-    std::string base_url = REALM_QUOTE(REALM_MONGODB_ENDPOINT);
-    if (base_url.size() > 0 && base_url[0] == '"') {
-        base_url.erase(0, 1);
-    }
-    if (base_url.size() > 0 && base_url[base_url.size() - 1] == '"') {
-        base_url.erase(base_url.size() - 1);
-    }
-    return base_url;
+    return unquote_string(REALM_QUOTE(REALM_MONGODB_ENDPOINT));
```

Comment:
are we quoting and then unquoting the string?

Reply:
I believe the original code was trying to "normalize" the string by removing the possible leading & trailing quotes, in the case where `REALM_MONGODB_ENDPOINT="\"http://127.0.0.1:9090\""` when this value is set during compilation.

Reply:
So is it really needed?
It doesn't matter on macos/clang at least if i configure with -DREALM_MONGODB_ENDPOINT=http://localhost:9090 or -DREALM_MONGODB_ENDPOINT="http://localhost:9090"
The variable in CMakeCache.txt is exactly the same value which doesn't need REALM_QUOTE at all since it's provided as a string through compile definitions.
Can we remove this function and REALM_QUOTE define altogether?

Reply:
I would imagine there was a reason to add this support when the endpoint value was first implemented. Maybe different platforms, build systems behave differently?
All I did was to pull it out into a separate function so I could reuse it for the admin endpoint.
I would prefer to leave as-is, in case it is required.